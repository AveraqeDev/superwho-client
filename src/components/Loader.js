import React from 'react';

import '../styles/Loader.css';

const Loader = () => {
  return (
    <div className="Loader">
      <div className="Loader__bounce1" />
      <div className="Loader__bounce2" />
      <div className="Loader__bounce3" />
    </div>
  );
}

export default Loader;