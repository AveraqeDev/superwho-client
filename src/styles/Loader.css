.Loader {
  position: absolute;
  top: 50%;
  left: 50%;
  width: 150px;
  text-align: center;
}

.Loader > div {
  width: 20px;
  height: 20px;
  background-color: #2A9BE4;

  margin-left: 10px;

  border-radius: 100%;
  display: inline-block;
  -webkit-animation: sk-bouncedelay 1.4s infinite ease-in-out both;
  animation: sk-bouncedelay 1.4s infinite ease-in-out both;
}

.Loader > div:nth-child(1) {
  animation-delay: -0.32s;
}

.Loader > div:nth-child(2) {
  animation-delay: -0.16s;
}

@-webkit-keyframes sk-bouncedelay {
  0%, 80%, 100% {
    -webkit-transform: scale(0);
  }
  40% {
    -webkit-transform: scale(1.0);
  }
}

@keyframes sk-bouncedelay {
  0%, 80%, 100% {
    -webkit-transform: scale(0);
    transform: scale(0);
  }
  40% {
    -webkit-transform: scale(1.0);
    transform: scale(1.0);
  }
}