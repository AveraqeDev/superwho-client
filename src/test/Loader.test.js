import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Loader from '../components/Loader';

describe('<Loader />', () => {
  // Smoke test
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Loader />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  //Snapshot test
  it('renders the UI as expected', () => {
    const tree = renderer
      .create(
        <Loader />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  })
});