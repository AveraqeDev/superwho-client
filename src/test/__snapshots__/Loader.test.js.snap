// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> renders the UI as expected 1`] = `
<div
  className="Loader"
>
  <div
    className="Loader__bounce1"
  />
  <div
    className="Loader__bounce2"
  />
  <div
    className="Loader__bounce3"
  />
</div>
`;
